src/rendering.ts

```diff
@@ -578,8 +578,14 @@ export class StatusmapRenderer {
     if (this.panel.color.mode === 'opacity') {
       return this.panel.color.cardColor;
     } else if (this.panel.color.mode === 'spectrum') {
-      return this.colorScale(bucket.value);
-    } else if (this.panel.color.mode === 'discrete') {
+      if(bucket.value !== null) {
+        return this.colorScale(bucket.value);
+      } else {
+          if(this.panel.nullPointMode == 'as zero')  {
+            return this.colorScale(0);
+          }
+        }
+      } else if (this.panel.color.mode === 'discrete') {
```

Comment:
Hello! Thank you for your contribution!
I've run a plugin with the suggested change and it not work. Check for null is not working, because `bucket.value` may be "undefined". I've changed your code a bit and it works. Can you apply this suggestion?

```suggestion
      if (!bucket.value && this.panel.nullPointMode === 'as zero') {
        // bucket.value === 0 falls here, but it is fine.
        return this.colorScale(0);
      }
      return this.colorScale(bucket.value);
    } else if (this.panel.color.mode === 'discrete') {
```